pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -303,6 +313,20 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
+
+            if optimize_sync:
+                optimize_data = dict(message='Optimizing Sync', code='optimizing.zync')
+                with ProgressReport(**optimize_data) as optimize_pb:
+                    if repomd.revision <= self.repository.revision_number:
+                        if self.remote.pulp_last_updated == self.repository.remote_last_updated:
+                            if self.repository.last_sync_repo_version ==\
+                                    (DeclarativeVersion.new_version - 1):
```

Comment:
Could you elaborate why DeclarativeVersion.new_version is used here?
It looks to me like a class attribute/method, and I'm not sure how  the current repository version number can be there. Maybe I miss something.

Reply:
This is just a placeholder at the moment as I don't know how to get the info I want here.  I was intending to ask for suggestions next time we discussed it...so...suggestions? :)

Reply:
```suggestion
                                    (self.repository.next_version - 1):
```
or 
```suggestion
                                    (self.repository.latest_version().number):
```

Reply:
`next_version` stores what would be the number of the next version,
[latest_version](https://github.com/pulp/pulpcore/blob/master/pulpcore/app/models/repository.py#L133-L146) gets the latest RepositoryVersion